### PR TITLE
Linter: check the license_family is valid

### PR DESCRIPTION
**Pull request by kynan**
conda-build actually checks for that. Relates to #88.

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -86,7 +88,13 @@ def lintify(meta, recipe_dir=None):
     if 'unknown' == license.strip():
         lints.append('The recipe license cannot be unknown.')
 
-    # 6: Selectors should be in a tidy form.
+    # 6: License family must be valid (conda-build checks for that)
```
Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: I don't know that we want to be messing with the number. Maybe @pelson can comment.

**Comment by jakirkham** [comment]
It's a bit redundant as `conda-build` does check it. That being said, am in favor of not having to hunt through logs for information like this. So having the linter say it is an improvement.

**Comment by kynan** [comment]
I agree it's a redundant check, but it will be detected at lint rather than build time.

File: conda_smithy/lint_recipe.py (hunk 1 above)
Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: Let's go ahead and put it at the end. May need to rebase though as I just merged PR ( https://github.com/conda-forge/conda-smithy/pull/334 ).

**Comment by kynan** [comment]
@jakirkham done!

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -122,6 +124,12 @@ def lintify(meta, recipe_dir=None):
         lints.append('The recipe `license` should not include the word '
                      '"License".')
 
+    # 11: License family must be valid (conda-build checks for that)
+    license_family = about_section.get('license_family', '')
+    if license_family and not license_family in allowed_license_families:
+        lints.append('The recipe license_family `{}` is invalid: must be one of {}.'
+                     ''.format(license_family, ', '.join(allowed_license_families)))
```
Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: Actually why don't we just use `ensure_valid_license_family` to verify this?

Comment (jakirkham) [inline conda_smithy/lint_recipe.py]: That's certainly an option and should be easy to do. 👍

**Comment by kynan** [comment]
@jakirkham now using `ensure_valid_license_family` directly. The build was cancelled, can't tell why.

**Comment by jakirkham** [comment]
> The build was cancelled, can't tell why.

Weird. 😕 Restarted and seems to have worked.

However, there appears to be a merge conflict. Likely due to another linter change.

**Comment by kynan** [comment]
@jakirkham rebased on upstream master!

**Pull request merged**